File: src (cpp)/animal/reptilia/crocodile.cpp
#include "crocodile.h"

Crocodile::Crocodile(int _x, int _y, int _weight) : def_ID(8),
                                                    def_ratio_meat(60),
                                                    def_ratio_plant(0) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = true;
  is_water_animal = true;
  is_air_animal = false;
  weight = _weight;
}
void Crocodile::Interact() {
  std::cout << "The crocodile is laying still on a slab of rock" << std::endl;
}

File: src/virtualzoo/animal/reptilia/Crocodile.java
package virtualzoo.animal.reptilia;

import virtualzoo.animal.behavior.BehaviorTame;
import virtualzoo.animal.diet.Carnivora;

/**
 * Kelas Crocodile mendefinisikan atribut untuk crocodile.
 *
 * @author Rionaldi Chandraseta - 13515077
 * @version 3.0
 * @since 3.0
 */
public class Crocodile extends Reptilia {

  /** <p>
   * Constructor
   *
   * Menciptakan crocodile dengan posisi (locX,locY) dan berat animalWeight.
   * </p>
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat crocodile.
   */
  public Crocodile(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorTame();
    animalDiet = new Carnivora(animalWeight, 60);
    id = 8;
    position.setLocation(locX, locY);
    isLandAnimal = true;
    isWaterAnimal = true;
    isAirAnimal = false;
  }

  /**
   * Mengatur interaksi dengan crocodile.
   */
  public void interact() {
    System.out.println("The crocodile is laying still on a slab of rock");
  }
}
